shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/alter/excutor/AlterTrafficRuleExecutor.java

```diff
@@ -0,0 +1,124 @@
+                result.add(each.getLoadBalancer().getName());
+            }
+        });
+        return result;
+    }
+    
+    private TrafficRuleConfiguration createTrafficRuleConfiguration(final AlterTrafficRuleStatement sqlStatement) {
+        TrafficRuleConfiguration result = new TrafficRuleConfiguration();
+        sqlStatement.getSegments().forEach(each -> setConfigurationData(result, each));
+        return result;
+    }
+    
+    private void setConfigurationData(final TrafficRuleConfiguration result, final TrafficRuleSegment each) {
+        ShardingSphereAlgorithmConfiguration trafficAlgorithm = createAlgorithmConfiguration(each.getAlgorithm());
+        ShardingSphereAlgorithmConfiguration loadBalancer = createAlgorithmConfiguration(each.getLoadBalancer());
+        String trafficAlgorithmName = createAlgorithmName(each.getName(), trafficAlgorithm);
+        String loadBalancerName = createAlgorithmName(each.getName(), loadBalancer);
+        TrafficStrategyConfiguration trafficStrategy = createTrafficStrategy(each, trafficAlgorithmName, loadBalancerName);
+        result.getTrafficStrategies().add(trafficStrategy);
+        result.getTrafficAlgorithms().put(trafficAlgorithmName, trafficAlgorithm);
+        result.getLoadBalancers().put(loadBalancerName, loadBalancer);
+    }
+    
+    private ShardingSphereAlgorithmConfiguration createAlgorithmConfiguration(final AlgorithmSegment segment) {
+        return new ShardingSphereAlgorithmConfiguration(segment.getName(), segment.getProps());
+    }
+    
+    private TrafficStrategyConfiguration createTrafficStrategy(final TrafficRuleSegment trafficRuleSegment, final String trafficAlgorithmName, final String loadBalancerName) {
+        return new TrafficStrategyConfiguration(trafficRuleSegment.getName(), trafficRuleSegment.getLabels(), trafficAlgorithmName, loadBalancerName);
+    }
+    
+    private void updateToRepository(final TrafficRuleConfiguration toBeAlteredConfiguration, final TrafficRuleConfiguration currentConfiguration) {
+        currentConfiguration.getTrafficStrategies().addAll(toBeAlteredConfiguration.getTrafficStrategies());
+        currentConfiguration.getTrafficAlgorithms().putAll(toBeAlteredConfiguration.getTrafficAlgorithms());
+        currentConfiguration.getLoadBalancers().putAll(toBeAlteredConfiguration.getLoadBalancers());
+        MetaDataContexts metaDataContexts = ProxyContext.getInstance().getContextManager().getMetaDataContexts();
+        Optional<MetaDataPersistService> metaDataPersistService = metaDataContexts.getMetaDataPersistService();
+        metaDataPersistService.ifPresent(op -> op.getGlobalRuleService().persist(metaDataContexts.getGlobalRuleMetaData().getConfigurations(), true));
+    }
+    
+    private String createAlgorithmName(final String ruleName, final ShardingSphereAlgorithmConfiguration algorithm) {
```

Comment:
The method `createAlgorithmName` is existed in `CreateTrafficRuleHandler`,  can we reuse it?  By adding a public class?